juju/juju
# pass regionSpec for regionInheritedConfig in model-cfg

This is part of the mode-config-tree work. It fixes acquisition of
region attributes in models running in regions different than the
controller.

QA: 
1. import the cloud at; http://paste.ubuntu.com/23080517/
2. juju bootstrap reed awstest
3. juju add-model west  && juju model-config # verify no-proxy: https://foo-west
4. juju add-model east --region us-east-1 && juju model-config # verify no-proxy: https://foo-east
5. juju add-model west2 --region us-west-2 && juju model-config # verify no-proxy: https://local
6. Unit tests all pass.

(Review request: http://reviews.vapour.ws/r/5510/)
